Automattic/wp-calypso
# Purchases: Add notice that purchases may use backup if available

#### Changes proposed in this Pull Request

This adds small text notices to the purchase list and details pages if there is a backup payment method set.

Each notice is a link to the payments documentation on backup cards (similar to the links that are added by https://github.com/Automattic/wp-calypso/pull/56665). Note that the actual updated docs are not yet published, but the link should still work.

#### Screenshots

The subscription list page:
<img width="1031" alt="Screen Shot 2021-11-30 at 4 42 01 PM" src="https://user-images.githubusercontent.com/2036909/144132923-4d9deba1-875d-4d48-b71f-013713a8741e.png">

The subscription details page:
<img width="1046" alt="Screen Shot 2021-11-30 at 4 42 41 PM" src="https://user-images.githubusercontent.com/2036909/144132937-bfc239df-6553-4266-bad4-b09ba5c6b88d.png">


#### Testing instructions

- Make sure that no payment methods are marked as backup (you can use Payments Admin).
- Visit `/me/purchases` and verify that you do not see any notice of backup payment methods.
- Click on a purchase and verify that you do not see any notice of backup payment methods.

Then:

- Assign at least one payment method as a backup.
- Repeat the above steps and verify that you see a notice on both pages.
